extractor/filesystem/language/python/pdmlock/extractor_test.go

```diff
@@ -0,0 +1,235 @@
+package pdmlock_test
+
+import (
+	"testing"
+
+	"github.com/google/osv-scalibr/extractor"
+	"github.com/google/osv-scalibr/extractor/filesystem/language/python/pdmlock"
+	"github.com/google/osv-scalibr/extractor/filesystem/osv"
+	"github.com/google/osv-scalibr/testing/sharedtesthelpers"
+)
+
+func TestPdmExtractor_FileRequired(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name        string
+		inputConfig sharedtesthelpers.ScanInputMockConfig
+		want        bool
+	}{
+		{
+			name: "empty",
+			inputConfig: sharedtesthelpers.ScanInputMockConfig{
+				Path: "",
+			},
+			want: false,
+		},
+		{
+			name: "plain",
+			inputConfig: sharedtesthelpers.ScanInputMockConfig{
+				Path: "pdm.lock",
+			},
+			want: true,
+		},
+		{
+			name: "absolute",
+			inputConfig: sharedtesthelpers.ScanInputMockConfig{
+				Path: "/path/to/pdm.lock",
+			},
+			want: true,
+		},
+		{
+			name: "relative",
+			inputConfig: sharedtesthelpers.ScanInputMockConfig{
+				Path: "../../pdm.lock",
+			},
+			want: true,
+		},
+		{
+			name: "in-path",
+			inputConfig: sharedtesthelpers.ScanInputMockConfig{
+				Path: "/path/with/pdm.lock/in/middle",
+			},
+			want: false,
+		},
+		{
+			name: "invalid-suffix",
+			inputConfig: sharedtesthelpers.ScanInputMockConfig{
+				Path: "pdm.lock.file",
+			},
+			want: false,
+		},
+		{
+			name: "invalid-prefix",
+			inputConfig: sharedtesthelpers.ScanInputMockConfig{
+				Path: "project.name.pdm.lock",
+			},
+			want: false,
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			e := pdmlock.Extractor{}
+			got := e.FileRequired(tt.inputConfig.Path, sharedtesthelpers.GenerateFileInfoMock(t, tt.inputConfig))
```

Comment:
I'm not sure if we really need to generate a whoel ScanInputConfig for the IsRequired tests. Can't we just pass the Path directly and have a separate helper function for creating a "normal" 
FileInfo?

Reply:
I think this whole mock thing is added for the go binary extractor which needs to know FileMode, but I agree we don't need to apply it to everything, I'll change it to use a normal path.